Tools/TestWebKitAPI/Tests/WTF/cf/RetainPtr.cpp

```diff
@@ -181,4 +181,18 @@ TEST(RetainPtr, RetainPtrCF)
     EXPECT_EQ(1, CFGetRetainCount(object2.get()));
 }
 
+static void functionWithRetainedCFOutParam(CFNumberRef* __nonnull CF_RETURNS_RETAINED retainedOutParam)
+{
+    float value = 3.1415926535;
+    CFNumberRef number = CFNumberCreate(kCFAllocatorDefault, kCFNumberFloatType, &value);
+    *retainedOutParam = number;
+}
+
+TEST(RetainPtr, RetainedOutParams)
+{
+    RetainPtr<CFNumberRef> value;
+    functionWithRetainedCFOutParam(adoptOutParam(value));
```

Comment:
So why do we need the `RetainPtrGetterRetains<T>` helper class?  Is that just to make the syntax work for `adoptOutParam(value)`?

Because I think we could just rename `RetainPtr<T>::internalValuePointer()` to `RetainPtr<T>::adoptOutParam()` and get the same effect without the  `RetainPtrGetterRetains<T>` class by doing this:

```
    RetainPtr<CFNumberRef> value;
    functionWithRetainedCFOutParam(value.adoptOutParam());
```

Reply:
I considered this, but thought the `adoptOutParam(value)` pattern was better; it also hides the RetainPtr function that exposes a pointer to the value.